File: CS/04_Graphs/Overlay.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;


namespace Overlay
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Load two documents from disk
            PdfDocument doc1 = new PdfDocument();
            doc1.LoadFromFile(@"..\..\..\..\..\..\Data\Overlay1.pdf");

            PdfDocument doc2 = new PdfDocument();
            doc2.LoadFromFile(@"..\..\..\..\..\..\Data\Overlay2.pdf");

            //Create page template form the first page of doc1
            PdfTemplate template = doc1.Pages[0].CreateTemplate();

            //Iterate each page in doc2
            foreach (PdfPageBase page in doc2.Pages)
            {
                //Set transparency for page 
                page.Canvas.SetTransparency(0.25f, 0.25f, PdfBlendMode.Overlay);

                //Draw template
                page.Canvas.DrawTemplate(template, PointF.Empty);
            }

            //Save the document
            doc2.SaveToFile("Overlay.pdf");
            doc1.Close();
            doc2.Close();

            //Launch the Pdf file
            PDFDocumentViewer("Overlay.pdf");
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/04_Graphs/Overlay.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace Overlay
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object
			Dim doc1 As New PdfDocument()

			' Load the PDF document from the specified file path
			doc1.LoadFromFile("..\..\..\..\..\..\Data\Overlay1.pdf")

			' Create a new PdfDocument object
			Dim doc2 As New PdfDocument()

			' Load the PDF document from the specified file path
			doc2.LoadFromFile("..\..\..\..\..\..\Data\Overlay2.pdf")

			' Create a PdfTemplate from the first page of doc1
			Dim template As PdfTemplate = doc1.Pages(0).CreateTemplate()

			' Iterate through each page in doc2
			For Each page As PdfPageBase In doc2.Pages
				' Set the transparency of the page's canvas using overlay mode
				page.Canvas.SetTransparency(0.25F, 0.25F, PdfBlendMode.Overlay)

				' Draw the template on the page's canvas at the top-left corner
				page.Canvas.DrawTemplate(template, PointF.Empty)
			Next page

			' Save the modified doc2 as Overlay.pdf
			doc2.SaveToFile("Overlay.pdf")

			' Close the documents
			doc1.Close()
			doc2.Close()

			' Launch the Pdf file
			PDFDocumentViewer("Overlay.pdf")
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
